### PR TITLE
Refactored plugins to avoid code duplication

### PR DESCRIPTION
**Pull request by JohanMabille**

### CODE REVIEW

**Comment by jtpio** [comment]
Thanks @JohanMabille.

The code for the two plugins is indeed very similar. I'm just wondering if this is going to be the case later on.

File: src/handlers/console.ts [hunk 1]
```diff
@@ -14,9 +14,9 @@ import { IDisposable } from '@phosphor/disposable';
 import { Signal } from '@phosphor/signaling';
 
 export class DebuggerConsoleHandler implements IDisposable {
-  constructor(options: DebuggerNotebookHandler.IOptions) {
+  constructor(options: DebuggerConsoleHandler.IOptions) {
```
Comment (jtpio) [inline src/handlers/console.ts]: Nice catch :+1:

**Comment by JohanMabille** [comment]
Alright, then let's keep the `activate` methods of each plugin and refactor smaller parts of their implementation so we avoid some code duplication while keeping flexibility.

**Comment by jtpio** [comment]
That sounds like a good compromise :+1: 

We can also fix the `DebuggerNotebookHandler` namespace in the console handler (https://github.com/jupyterlab/debugger/pull/78#discussion_r337012964) in a separate PR.

**Comment by JohanMabille** [comment]
Ok, PR incoming shortly

**Comment by JohanMabille** [comment]
@jtpio we can go further in refactoring old setting handler, but that would require function with constructor generic (the setOldHandler function should accept type like `new(a: any): T` with `T = DebuggerConsoleHandler | DebuggerNotebookHandler`).

**Comment by JohanMabille** [comment]
@jtpio this is ready for review.

**Comment by jtpio** [comment]
Thanks @JohanMabille.

cc @afshin who might want to look at it.

**Pull request merged**